File: datasets/lua/HumanEval_28_concatenate.lua
-- Concatenate list of strings into a single string
-- >>> concatenate([])
-- ''
-- >>> concatenate(['a', 'b', 'c'])
-- 'abc'
local function concatenate(strings)
  local result = ''
  for _, str in ipairs(strings) do
    result = result .. str
  end
  return result
end

lu = require('luaunit')

function test_humaneval()
local candidate = concatenate
    lu.assertEquals(candidate({}), '')
    lu.assertEquals(candidate({'x', 'y', 'z'}), 'xyz')
    lu.assertEquals(candidate({'x', 'y', 'z', 'w', 'k'}), 'xyzwk')
end

os.exit(lu.LuaUnit.run())

File: datasets/rs/HumanEval_28_concatenate.rs
/// Concatenate list of strings into a single string
/// >>> concatenate([])
/// ''
/// >>> concatenate(['a', 'b', 'c'])
/// 'abc'
fn concatenate(strings: Vec<String>) -> String {
    let mut result = String::new();
    for s in strings {
        result.push_str(&s);
    }
    result

}

fn main() {
    let candidate = concatenate;
    assert_eq!(candidate(Vec::<String>::new()), String::from(""));
    assert_eq!(candidate(vec![String::from("x"), String::from("y"), String::from("z")]), String::from("xyz"));
    assert_eq!(candidate(vec![String::from("x"), String::from("y"), String::from("z"), String::from("w"), String::from("k")]), String::from("xyzwk"));
}
